src/System.IO.FileSystem/src/System/IO/FileSystem.DirectoryCreation.Windows.cs

```diff
@@ -0,0 +1,132 @@
+namespace Microsoft.IO
+#else
+namespace System.IO
+#endif
+{
+    internal static partial class FileSystem
+    {
+        public static void CreateDirectory(string fullPath, byte[] securityDescriptor = null)
+        {
+            // We can save a bunch of work if the directory we want to create already exists.  This also
+            // saves us in the case where sub paths are inaccessible (due to ERROR_ACCESS_DENIED) but the
+            // final path is accessible and the directory already exists.  For example, consider trying
+            // to create c:\Foo\Bar\Baz, where everything already exists but ACLS prevent access to c:\Foo
+            // and c:\Foo\Bar.  In that case, this code will think it needs to create c:\Foo, and c:\Foo\Bar
+            // and fail to due so, causing an exception to be thrown.  This is not what we want.
+            if (DirectoryExists(fullPath))
+                return;
+
+            List<string> stackDir = new List<string>();
+
+            // Attempt to figure out which directories don't exist, and only
+            // create the ones we need.  Note that FileExists may fail due
+            // to Win32 ACL's preventing us from seeing a directory, and this
+            // isn't threadsafe.
+
+            bool somepathexists = false;
+
+            int length = fullPath.Length;
+
+            // We need to trim the trailing slash or the code will try to create 2 directories of the same name.
+            if (length >= 2 && Path.EndsInDirectorySeparator(fullPath.AsSpan()))
+                length--;
+
+            int lengthRoot = PathInternal.GetRootLength(fullPath.AsSpan());
+
+            if (length > lengthRoot)
+            {
+                // Special case root (fullpath = X:\\)
+                int i = length - 1;
+                while (i >= lengthRoot && !somepathexists)
+                {
+                    string dir = fullPath.Substring(0, i + 1);
+
+                    if (!DirectoryExists(dir)) // Create only the ones missing
+                        stackDir.Add(dir);
+                    else
+                        somepathexists = true;
+
+                    while (i > lengthRoot && !PathInternal.IsDirectorySeparator(fullPath[i])) i--;
+                    i--;
+                }
+            }
+
+            int count = stackDir.Count;
+            bool r = true;
+            int firstError = 0;
+            string errorString = fullPath;
+
+            unsafe
+            {
+                fixed (byte* pSecurityDescriptor = securityDescriptor)
+                {
+                    Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
+                    {
+                        nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES),
+                        lpSecurityDescriptor = (IntPtr)pSecurityDescriptor
+                    };
+
+                    // If all the security checks succeeded create all the directories
```

Comment:
nit- this comment is dead- can you remove it?

Reply:
Thanks for noticing. I removed it.